src/engine/strat_engine/filesystem.rs

```diff
@@ -64,6 +97,33 @@ impl StratFilesystem {
         self.thin_dev.id()
     }
 
+    /// Return an target size for the filesystem over the thin device.
+    /// TODO: returning 2 times the current size will double the space provisoned to
+    /// the filesystem.  We should determine if this is a reasonable value.
+    fn extend_size(&self, current_size: DataBlocks) -> DataBlocks {
```

Comment:
Not sure I follow.  They do the same thing, one for a filesystem - one for a pool.  don't we do similar stuff with teardown(), rename() etc?